basictracer/text_propagator.py

```diff
@@ -12,6 +12,34 @@
 field_count = 3
 
 
+def parse_hex_for_field(field_name, value):
+    """parses the hexadecimal value of a field into an integer.
+    Raises SpanContextCorruptedException in case of failure
+    """
+    msg = '{field_name} got an invalid hexadecimal value {value!r}'
+    msg = msg.format(field_name=field_name, value=value)
```

Comment:
no need to prepare this message if ValueError is not raised. Can be moved inside of `except` clause